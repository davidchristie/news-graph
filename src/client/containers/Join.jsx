import React, { PropTypes } from 'react'
import { connect } from 'react-redux'

import { signup } from '../actions/account'
import Alerts from '../components/Alerts'
import Jumbotron from '../components/Jumbotron'

export const Join = class Join extends React.Component {

  constructor (props) {
    super(props)
    this.inputs = {}
    this.handleSubmit = this.handleSubmit.bind(this)
    this.state = {alerts: []}
  }

  handleSubmit (event) {
    event.preventDefault()
    const form = event.target
    const dispatch = this.props.dispatch
    const name = this.inputs.name.value
    const email = this.inputs.email.value
    const password = this.inputs.password.value
    const password2 = this.inputs.password2.value
    dispatch(signup({email, name, password, password2}))
      .then(response => {
        this.setState({alerts: response.alerts})
        if (response.success) {
          form.reset()
        } else {
          // window.scrollTo(0, 0)
        }
      })
  }

  render () {
    return (
      <div>
        <Jumbotron title='Join' />

        <div className='card'>
          <div className='card-block'>

            <Alerts alerts={this.state.alerts} />

            <form onSubmit={this.handleSubmit}>

              <div className='form-group'>
                <label htmlFor='join-name'>Name</label>
                <input id='join-name' type='text' className='form-control' placeholder='Name' name='name' required ref={input => { this.inputs.name = input }} />
              </div>

              <div className='form-group'>
                <label htmlFor='join-email'>Email</label>
                <input id='join-email' type='email' className='form-control' placeholder='Email address' name='email' required ref={input => { this.inputs.email = input }} />
              </div>

              <div className='form-group'>
                <label htmlFor='join-password'>Password</label>
                <input id='join-password' type='password' className='form-control' placeholder='Password' name='password' required ref={input => { this.inputs.password = input }} />
              </div>

              <div className='form-group'>
                <label htmlFor='join-password2'>Confirm Password</label>
                <input id='join-password2' type='password' className='form-control' placeholder='Re-type password' name='password2' required ref={input => { this.inputs.password2 = input }} />
              </div>

              <button type='submit' className='btn btn-outline-success'>Submit</button>

            </form>

          </div>
        </div>
      </div>
    )
  }

}

Join.propTypes = {
  dispatch: PropTypes.func.isRequired
}

export default connect()(Join)
