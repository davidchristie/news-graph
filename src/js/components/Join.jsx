import React from 'react'

import Jumbotron from './Jumbotron'

export default class About extends React.Component {
  render() {
    const title = 'Join'
    return (
      <div>
        <Jumbotron title={title} />

        <div className="card">
          <div className="card-block">

            <form method="post" action="/users/join">

              <div className="form-group">
                <label htmlFor="join-name">Name</label>
                <input id="join-name" type="text" className="form-control" placeholder="Name" name="name" required />
              </div>

              <div className="form-group">
                <label htmlFor="join-email">Email</label>
                <input id="join-email" type="email" className="form-control" placeholder="Email address" name="email" required />
              </div>

              <div className="form-group">
                <label htmlFor="join-password">Password</label>
                <input id="join-password" type="password" className="form-control" placeholder="Password" name="password" required />
              </div>

              <div className="form-group">
                <label htmlFor="join-password2">Confirm Password</label>
                <input id="join-password2" type="password" className="form-control" placeholder="Re-type password" name="password2" required />
              </div>

              <button type="submit" className="btn btn-outline-success">Submit</button>

            </form>

          </div>
        </div>
      </div>
    )
  }
}
