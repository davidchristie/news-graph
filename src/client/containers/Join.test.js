/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { Join } from './Join'

test('Join renders correctly', () => {
  const dispatch = () => {}
  const wrapper = shallow(<Join dispatch={dispatch} />)
  expect(toJson(wrapper)).toMatchSnapshot()
})
