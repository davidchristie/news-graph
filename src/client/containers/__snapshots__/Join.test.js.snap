exports[`test Join renders correctly 1`] = `
<div>
  <Jumbotron
    title="Join" />
  <div
    className="card">
    <div
      className="card-block">
      <Alerts
        alerts={Array []} />
      <form
        onSubmit={[Function]}>
        <div
          className="form-group">
          <label
            htmlFor="join-name">
            Name
          </label>
          <input
            className="form-control"
            id="join-name"
            name="name"
            placeholder="Name"
            required={true}
            type="text" />
        </div>
        <div
          className="form-group">
          <label
            htmlFor="join-email">
            Email
          </label>
          <input
            className="form-control"
            id="join-email"
            name="email"
            placeholder="Email address"
            required={true}
            type="email" />
        </div>
        <div
          className="form-group">
          <label
            htmlFor="join-password">
            Password
          </label>
          <input
            className="form-control"
            id="join-password"
            name="password"
            placeholder="Password"
            required={true}
            type="password" />
        </div>
        <div
          className="form-group">
          <label
            htmlFor="join-password2">
            Confirm Password
          </label>
          <input
            className="form-control"
            id="join-password2"
            name="password2"
            placeholder="Re-type password"
            required={true}
            type="password" />
        </div>
        <button
          className="btn btn-outline-success"
          type="submit">
          Submit
        </button>
      </form>
    </div>
  </div>
</div>
`;
